.github/workflows/ci-functional-perl.yml

```diff
@@ -0,0 +1,58 @@
+name: CI for NJS-based SAML Implementation
+run-name: ${{ github.actor }} is triggering pipeline
+on: [push]
+
+jobs:
+  test-njs-saml:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+  
+      - name: Install prerequisites
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y apt-transport-https lsb-release apt-utils ubuntu-keyring gnupg2 \
+          ca-certificates wget mercurial
+
+      - name: Prepare keys and certificates
+        run: |
+          sudo mkdir /etc/ssl/nginx
+          echo '${{ secrets.NGINX_REPO_CRT }}' | sudo tee /etc/ssl/nginx/nginx-repo.crt > /dev/null
```

Comment:
Why so interesting approach to out var content? may be better just to use: `printenv var_name > destination`

Reply:
The reason is that the directory /etc/ssl/nginx requires elevated privileges to write to.
Using `printenv`, we would face a challenge because the redirection operator `>` does not inherit the privileges granted by sudo.

Reply:
However, technically, we could consider using an alternative directory that is accessible to a non-privileged user.

Reply:
OK